presto-spi/src/main/java/com/facebook/presto/spi/relation/DefaultRowExpressionTranslator.java

```diff
@@ -0,0 +1,75 @@
+
+    public DefaultRowExpressionTranslator(ConnectorPlanOptimizerProvider.TranslationContext translationContext)
+    {
+        this.translator = translationContext;
+    }
+
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateRowExpression(RowExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return null;
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateInputReference(InputReferenceExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return translateRowExpression(node, arguments, context, treeRewriter);
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateCall(CallExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return (ConnectorPlanOptimizerProvider.TargetExpression<T>) this.translator.getFunctionTranslator().apply(node, arguments);
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateConstant(ConstantExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return translateRowExpression(node, arguments, context, treeRewriter);
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateLambda(LambdaDefinitionExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return translateRowExpression(node, arguments, context, treeRewriter);
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateVariableReference(VariableReferenceExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return translateRowExpression(node, arguments, context, treeRewriter);
+    }
+
+    @Override
+    public ConnectorPlanOptimizerProvider.TargetExpression<T> translateSpecialForm(SpecialFormExpression node, List<ConnectorPlanOptimizerProvider.TargetExpression<T>> arguments, ConnectorContext context, RowExpressionTreeTranslator<T, ConnectorContext> treeRewriter)
+    {
+        return translateRowExpression(node, arguments, context, treeRewriter);
+    }
+
+    public static class ConnectorContext
+    {
+        Map<VariableReferenceExpression, ColumnHandle> columnHandleMap;
```

Comment:
I wouldn't put this in spi. It is only used by translateVariableReference. We can remove this and ask connector to inherit DefaultRowExpressionTranslator